polaris-react/src/components/Navigation/components/Item/Item.tsx

```diff
@@ -203,29 +211,38 @@ export function Item({
     );
   }
 
-  const secondaryActionLinkMarkup = secondaryAction && (
-    <UnstyledLink
-      external
-      url={secondaryAction.url}
-      className={styles.SecondaryAction}
-      tabIndex={tabIndex}
-      aria-disabled={disabled}
-      aria-label={secondaryAction.accessibilityLabel}
-      onClick={secondaryAction.onClick}
-    >
-      <Icon source={secondaryAction.icon} />
-    </UnstyledLink>
-  );
+  if (secondaryAction && process.env.NODE_ENV === 'development') {
+    // eslint-disable-next-line no-console
+    console.warn(
```

Comment:
If you are able to replace the deprecated prop usage in `web` that would be super, to avoid unnecessary noise for devs

Reply:
Yup I'll be sure to do that post update in web.